manager/src/main/java/org/apache/hertzbeat/manager/config/PluginConfig.java

```diff
@@ -0,0 +1,177 @@
+            return null;
+        }
+        return Arrays.stream(files)
+                .filter(f -> f.getName().endsWith(".jar"))
+                .map(File::getAbsolutePath)
+                .collect(Collectors.toList());
+    }
+
+    private List<String> getAllClassNamesFromJar() throws IOException {
+        List<String> jars = getAllJars();
+        if (jars == null || jars.isEmpty()) {
+            return null;
+        }
+        List<String> classNames = new ArrayList<>();
+        for (String jar : jars) {
+            List<String> collect;
+            try (JarFile file = new JarFile(jar)) {
+                collect = file.stream()
+                        .filter(jarEntry -> jarEntry.getName().endsWith(CLASS_SUFFIX))
+                        .filter(jarEntry -> jarEntry.getName().startsWith(PACKAGE))
+                        .filter(jarEntry -> !jarEntry.getName().contains(BASE_PLUGIN_PATH))
+                        .map(jarEntry -> jarEntry.getName().replace("/", ".").replace(".class", ""))
+                        .toList();
+            }
+            classNames.addAll(collect);
+        }
+        return classNames;
+    }
+
+    private void registerBean(Class<?> c, BeanDefinitionRegistry registry) {
+        String className = c.getName();
+        BeanDefinitionBuilder builder = BeanDefinitionBuilder.genericBeanDefinition(c);
+        BeanDefinition beanDefinition = builder.getBeanDefinition();
+        registry.registerBeanDefinition(className, beanDefinition);
+    }
+
+    @Override
+    public void registerBeanDefinitions(@NotNull AnnotationMetadata importingClassMetadata, @NotNull BeanDefinitionRegistry registry, @NotNull BeanNameGenerator importBeanNameGenerator) {
+        URLClassLoader classLoader = getClassLoader();
+        List<String> allClassNamesFromJar;
+        try {
+            allClassNamesFromJar = getAllClassNamesFromJar();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        if (allClassNamesFromJar != null) {
+            for (String name : allClassNamesFromJar) {
+                if (name == null || name.trim().isEmpty()) {
+                    log.error("class name is null or empty");
+                    continue;
+                }
+                if (classLoader != null) {
+                    Class<?> clazz = null;
+                    try {
+                        clazz = classLoader.loadClass(name);
+                    } catch (ClassNotFoundException e) {
+                        log.error("load class error:{}", e.getLocalizedMessage());
+                    }
+                    if (clazz != null) {
+                        registerBean(clazz, registry);
+                    }
+                }
+            }
+        }    }
+
+    public List<Object> getBean() throws IOException {
+        List<String> names = getAllClassNamesFromJar();
+        List<Object> beans = new ArrayList<>();
+        if (names != null) {
```

Comment:
The enhanced for loop may no longer need to check whether `name != null`.

Reply:
Please help me modify it.

Reply:
This is my mistake, but it is still recommended to use `Optional` to reduce some conditional judgments.

```java
Optional.ofNullable(beans). ifPresent(()->{
// dosomthing
})

// or retrun Optional
Optional<List<String>> names = getAllClassNamesFromJar()
```